### DISCUSSION TITLE
AIS-catcher / no Internet

### DISCUSSION THREAD

**Discussion by OE1MWW** [jvde-github/AIS-catcher#152] [Ideas]
Hi all,
I am planning to carry my laptop + RTL-SDR on a cruise ship.
There is no Internet - oh yes, there is Internet, but it will cost 'an arm & a leg'

The browser page 'localhost:8100' can be called. OK, no internet, no map.
The 'Summary' page shows all received statistics. Unfortunately the remaining 'Plots' and 'Ships'
do not show any data at all. Is there a command line switch to enable this data - or is that state
of the server 'by design' ?

**Comment by jvde-github** [comment]
Hi, that is by design to keep the package light and simple to set up and use. But the drawback is that 3rd party libraries are fetched from the internet. 

I have been thinking a little bit about this. The Android app does have the same problem if you do not have a data connection. 

One thing that worked for me (but that is not guaranteed) is to run the web page when connected. I noticed that all libraries and a lot of map tiles where still in cache and available when offline. This was on a phone which I only used for this but the program ran fine. Unfortunately, you don't have control over the caching so that is not guaranteed and if you do a lot of other web things, it might be removed from cache.

Another option is to install OpenCPN which can use offline maps. 

Open to other suggestions from others?

**Comment by OE1MWW** [comment]
Jasper, thanks for the info. Yes, the only (ugly) option is OpenCPN.
Still missing the 'Ships' table with all the info of the web of AIS-catcher.

greetings, Wolfgang

**Reply by tkreyche** [reply]
Hi Wolfgang - I am curious why you consider OpenCPN an (ugly) option. I haven't used it but just started playing around with SignalK and am considering adding charting software. It appears there are several available besides OpenCPN.

thanks, Tom

**Comment by Taxom** [comment]
OpenCPN has a list of ships
![изображение](https://github.com/jvde-github/AIS-catcher/assets/20313019/cbad1952-3c8a-44c3-94d1-69d2e48613f6)

**Comment by OE1MWW** [comment]
Taxom,
thanks, I found it in the meantime too ;-)
Just have to findout how to set 'own_ship' (moving) in OpenCPN

**Comment by Taxom** [comment]
Depending on what you want, you can move your own ship manually:
![изображение](https://github.com/jvde-github/AIS-catcher/assets/20313019/02b6367e-53d5-49c3-b12b-1adff3d52150)

Or connect GPS through the connection menu:
![изображение](https://github.com/jvde-github/AIS-catcher/assets/20313019/d26bc625-b476-421e-9c5f-e193666fa59f)

**Comment by jvde-github** [comment]
FWIW, I just uploaded a new version for AIS-Catcher for Android that has now most functionality also offline (table, charts etc).

**Comment by jvde-github** [comment]
Ok, I created an experimental offline version as well. This should run normally but map tiles are not stored offline so that might work only depending what is in the cache. Graphs and tables work.

You first  go to your home directory (say `/home/jasper`) and clone the necessary web assets:
```console
git clone https://github.com/jvde-github/webassets.git
```
And then run AIS-catcher with the CDN switch for the webserver pointing to the created directory:
```console
AIS-catcher -x 192.168.1.120 4002 -N 8100 CDN /home/jasper/webassets
```

**Reply by dreua** [reply]
As a somewhat lazy person I'd love to see a ready built image of this :wink:

**Reply by jvde-github** [reply]
Good idea. I have added it to the new installation script, so, I think it is now fairly simple: Just install the raspberry image and then run the install script and you should be good to go,.

**Comment by OE1MWW** [comment]
Jasper, thanks for the new version, downloaded it
- did the git clone... to 100%
- inserted the CDN with the path in the command line
- it reports at start: "Fetch Web Libraries locally at D:/Temp/AISCatcher"

You call it the 'Edge version' (?) I am using Firefox, but tried the Windows Edge browser too, there is
no difference. Started the AISCatcher for a while without CDN, terminated and next started with the 
CDN parameter to fill cache etc.

'Summary' page is counting and filled with data, 'Plots' showing header lines, nothing more
'Ships' and 'Map' shows a blank white page.

**Reply by jvde-github** [reply]
CDN needs to point to a directory that has the CDN sub-folder. So if I clone webassets it will create a webassets directory somewhere (in my case I cloned in C:\Users\jvde\source\repos), that is the directory we need to specify (without a \ at the end). For me this works:

```console
AIS-catcher -x 192.168.1.239 4002 -N 8100 CDN "C:\Users\jvde\source\repos\webassets"
```

I will see if I can do a sense check at startup so that the user can get a more informative message. Will do that tonight or over the weekend.

![image](https://github.com/jvde-github/AIS-catcher/assets/52420030/38b3bd3a-4572-4a69-a050-afdba1d08447)

**Comment by OE1MWW** [comment]
I created an AISCatcher directory in my d:\Temp and started git in that directory
Got a 'webassets' directory and in this are subs '.git' and 'cdn'

So, I changed the pointer in start.cmd to "CDN D:/Temp/AISCatcher/webassets"
and that one works :-)

**Reply by jvde-github** [reply]
Travels can begin now :-)

**Reply by OE1MWW** [reply]
you saved my mood & time ;-)
wife: look, there is a ship
me: yes, I see
she: what ship is it, where does it go, how big is it, how fast does it go...
next time: "AISCatcher says - ships name is, speed is, heading is...
... and no more questions ;-) 

Thanks for your time!

**Comment by jnovos** [comment]
I think it is possible to download map areas for offline use and store them in an indexedBD. I have already made it work with LocalStorage but the storage size is tiny.

**Comment by NR857** [comment]
Just reviewing  these discussions as I am new to the group. 

I use Garmin/Navionics which allows you to download maps for offline use and works very well with Boat Beacon and/AIS Catcher. All you need to do is add devices and enable AIS targets. The sub for a years maps is about EUR 30 if I remember properly but you get chart updates, ATON, downloadable maps etc. 

Boat Beacon allows you to upload data for EUR 5 per month and again the functionality is excellent. 

I am still having difficulty networking to Navionics on a different device on the same WiFi network, but I think this is port conflicts or and/or NMEA talker/listener limits.

**Reply by jvde-github** [reply]
Hey, thanks for sharing. What protocol does navionics use?

**Reply by NR857** [reply]
UDP and_or TCP

**Reply by dreua** [reply]
Can you send data from AIS Catcher directly to Navionics and if so how? Or do you need the Boat Beacon app as kind of a proxy?

**Reply by jvde-github** [reply]
Just tried it on the demo version of navionics. I am running AIS-catcher on a computer with -S 5012. This computer has IP 192.168.1.120 on my local network.

Then in navionics boating app I go to paired decices and add a device with host 192.168.1.120 and port 5012 and protocol set to TCP. This seems to work but it gets disconnected after a while.

**Reply by dreua** [reply]
Thanks for the super fast reply, that was much easier than I thought.
I've been running it successfully on a raspberry in hotspot mode which does not provide internet. It works as soon as I disabled mobile data on my phone, no disconnections that I'm aware of. (Maybe after a long time in screen timeout or depending on your power saving settings it may kill the app.)

**Comment by NR857** [comment]
I presume you mean UDP or TCP?

You can select either. See attached screenshots of
